extensions/ql-vscode/src/view/model-editor/MethodRow.tsx

```diff
@@ -138,34 +152,46 @@ const ModelableMethodRow = forwardRef<HTMLElement | undefined, MethodRowProps>(
         )}
         {!props.modelingInProgress && (
           <>
-            <VSCodeDataGridCell gridColumn={2}>
-              <ModelTypeDropdown
-                method={method}
-                modeledMethod={modeledMethod}
-                onChange={onChange}
-              />
-            </VSCodeDataGridCell>
-            <VSCodeDataGridCell gridColumn={3}>
-              <ModelInputDropdown
-                method={method}
-                modeledMethod={modeledMethod}
-                onChange={onChange}
-              />
-            </VSCodeDataGridCell>
-            <VSCodeDataGridCell gridColumn={4}>
-              <ModelOutputDropdown
-                method={method}
-                modeledMethod={modeledMethod}
-                onChange={onChange}
-              />
-            </VSCodeDataGridCell>
-            <VSCodeDataGridCell gridColumn={5}>
-              <ModelKindDropdown
-                method={method}
-                modeledMethod={modeledMethod}
-                onChange={onChange}
-              />
-            </VSCodeDataGridCell>
+            <MultiModelColumn gridColumn={2}>
+              {forEachModeledMethod(modeledMethods, (modeledMethod) => (
+                <ModelTypeDropdown
+                  key={JSON.stringify(modeledMethod)}
```

Comment:
I think it might actually make sense here to use the index of the item in the array, even though this is not recommended in general. I'm just not sure if the JSON value of the modeled method is a better key since this may lead to duplicates when you create two models with exactly the same values. This may get React even more confused than using an index.

Reply:
Thanks for picking up on this. This was a bit of a hack to get the PR finished and I'd forgotten I did this 🙈 . I agree using the JSON value is not ideal.

Maybe using the index is fine in this case because we generally do identify modelings by their index rather than their content.

I've done a quick read of how react keys behave and my understanding is that using the JSON value like this would likely lead to unmounting/mounting a new component for every render. If we use indexes, then it would match up components by their index, and then still see the props have maybe changed and re-render them. But a re-render is much cheaper than creating and mounting a new component.